src/ImageSharp.Web/CaseHandlingUriBuilder.cs

```diff
@@ -121,19 +149,18 @@ public static string Encode(CaseHandling handling, Uri uri)
             {
                 return BuildAbsolute(
                     handling,
+                    scheme: uri.Scheme,
                     host: HostString.FromUriComponent(uri),
                     pathBase: PathString.FromUriComponent(uri),
                     query: QueryString.FromUriComponent(uri));
             }
             else
             {
-                string components = uri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
-                if (handling == CaseHandling.LowerInvariant)
-                {
-                    return components.ToLowerInvariant();
-                }
-
-                return components;
+                Uri faux = new(FallbackBaseUri, uri);
+                return BuildRelative(
+                    handling,
+                    pathBase: PathString.FromUriComponent(faux),
```

Comment:
Make more sense to set the `path`, as the `pathBase` is used as a prefix in ASP.NET Core (and by default also `string.Empty`:
```suggestion
                    path: PathString.FromUriComponent(faux),
```

Reply:
Yeah good point.